emission/net/ext_service/transit_matching/match_stops.py

```diff
@@ -96,17 +97,28 @@ def get_predicted_transit_mode(start_stops, end_stops):
         return [rim.tags.route for rim in rel_id_matches]
 
     # Did not find matching routes. Let's see if stops are both "railway",
-    # if so, we can mark as TRAIN 
+    # if so, we can mark as TRAIN
     # TODO: return more complex kinds of railways?
-    p_start_train = ["railway" in s.tags for s in start_stops]
-    p_end_train = ["railway" in s.tags for s in end_stops]
-
-    logging.debug("len(start_train) = %s, len(end_train) = %s" % (
-        (len(p_start_train), len(p_end_train))))
-    if is_true(p_start_train) and is_true(p_end_train):
-        logging.debug("start and end are both TRAIN, returning TRAIN")
-        return ["TRAIN"]
-
+    p_start_train = [extract_railway_modes(s.tags) for s in start_stops]
+    p_start_train = Counter(itertools.chain.from_iterable(set(i) for i in p_start_train))
```

Comment:
Again, for my own understanding, we have a set of tags for each potential start stop, and the `from_iterable` converts them into one giant list which you then use `Counter` over. I should really use `itertools` more instead of manually invoking list comprehensions all the time :)